gremlin-dotnet/src/Gremlin.Net/Structure/IO/GraphSON/GraphSONWriter.cs

```diff
@@ -148,7 +148,9 @@ private IGraphSONSerializer TryGetSerializerFor(Type type)
 
         private bool IsDictionaryType(Type type)
         {
-            return type.IsConstructedGenericType && type.GetGenericTypeDefinition() == typeof(Dictionary<,>);
+            return type
+                .GetInterfaces()
+                .Any(@interface => @interface.IsConstructedGenericType && @interface.GetGenericTypeDefinition() == typeof(IEnumerable<>) && @interface.GenericTypeArguments[0] is { IsConstructedGenericType: true } genericArgument && genericArgument.GetGenericTypeDefinition() == typeof(KeyValuePair<,>));
```

Comment:
(nitpick) Please split the long line.

Also how about renaming `@interface` to `interfaceType`? I think starting a variable name with `@` is a bit uncommon and it makes me wonder as a reader whether there's anything special to it to give it this name.

Reply:
Done.